sentry-android-core/src/main/java/io/sentry/android/core/SentryAndroid.java

```diff
@@ -15,6 +17,24 @@ private SentryAndroid() {}
    * @param context Application. context
    */
   public static void init(@NotNull final Context context) {
-    Sentry.init(options -> AndroidOptionsInitializer.init(options, context));
+    init(context, new AndroidLogger());
+  }
+
+  static void init(@NotNull final Context context, @NotNull ILogger logger) {
+    try {
+      Sentry.init(
+          SentryAndroidOptions.class,
+          options -> AndroidOptionsInitializer.init(options, context, logger));
+    } catch (IllegalAccessException e) {
+      // This is awful. Should we have this all on the interface and let the caller deal with these?
```

Comment:
Java has `Supplier` which helps on that, but only API 24

Reply:
Kotlin has `reified` btw.

quick question, why do we want to instantiate the options ourselves, would not be ok to allow them to instantiate their own `SentryAndroidOptions` and Sentry only get `<T extends SentryOptions>`? is there a reason to block it?

Reply:
This was discussed offline. tl;dr; Allow passing the options instance to framework callbacks that will read configuration data into the object (i.e: env var, xml config files etc) and only after that, provide the instance to the user callback for mutation.